### ISSUE TITLE
Breaking change in eslint@3.8.0 with babel-eslint parser (comma-dangle and rest object destructuring)

### ISSUE THREAD

**Issue by olalonde** [eslint/eslint#7422]
After updating from `eslint@3.7.0` to `eslint@3.8.0`, I am now suddenly getting errors whenever I have `...rest,` when destructuring an object, e.g.:

```javascript
const {
  prop1,
  ...rest, // Unexpected trailing comma  comma-dangle
} = obj
```

```
 error  Unexpected trailing comma  comma-dangle
```

If I revert back to `eslint@3.7.0` the error disappears.

Here's the [eslintrc I'm using](https://github.com/blockai/eslint-config-blockai):

```javascript
module.exports = {
  extends: 'airbnb-base',
  parser: 'babel-eslint',
  rules: {
    'consistent-return': 0,
    'no-param-reassign': [2, { props: false }],
    semi: [2, 'never'],
    camelcase: 0,
    'no-underscore-dangle': 0,
  },
}
```

**Comment by olalonde** [comment]
I suspect this might be because of this commit: https://github.com/eslint/eslint/commit/c8796e98af6804cccc947ab3e3b06d884a8d035f

I'm not sure if it should be fixed in babel-eslint or here.

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, I think this is intended behavior (caused by a bugfix in https://github.com/eslint/eslint/commit/52da71e9bfcdb7f13c6077531b992d059af7391a). Trailing commas after rest properties are a syntax error (see https://github.com/eslint/eslint/issues/7297 and https://github.com/eslint/eslint/pull/7298 for more info), so the solution is to remove the comma after `...rest` so that your syntax is valid.

**Comment by olalonde** [comment]
Ah right, doesn't really make sense to have a comma there since rest is alway last...
